src/masking/mod.rs

```diff
@@ -1 +1,91 @@
+use core::mem::MaybeUninit;
+use core::num::{NonZeroUsize, Wrapping};
+
+use crate::capacity::MaskingCapacity;
+use crate::storage::{PartialStorage, Storage};
+
 mod tests;
+
+pub struct MaskingRingBuffer<S: PartialStorage<Capacity = MaskingCapacity>> {
+    /// The underlying storage
+    storage: MaybeUninit<S>,
+    /// The start of the buffer in the storage (`0..CAPACITY`)
+    // TODO: Benchmark whether Wrapping is worth it
+    index: Wrapping<usize>,
```

Comment:
This field is always within range in your implementation (because you always mask it before writing this field); as such, `Wrapping` is just unhelpful.  It'd be useful if we did allow this field to overflow arbitrarily, and then only masked when accessing an element, but that's not the case here.

Reply:
Yeah I felt like over-masking for now. This is why I asked about `Wrapping`.